### DISCUSSION TITLE
Get response status even on "error"

### DISCUSSION THREAD

**Discussion by joshft91** [actions/github-script#386] [Q&A] [answered]
Hello! 👋 I wasn't sure the best phrase to word the question, but hopefully this makes sense.

I'm attempting to write a quick little function using `github-script` to `createOrUpdateRepoVariable` because the GitHub API only supports creating or patching repository variables and in my use-case, I won't know if that variable is already created or not yet.

Unfortunately, it doesn't look like `github.rest.actions.getRepoVariable` is implemented yet (#345) so I'm attempting to use `github.request('GET /...')` (shown in the example below) as a workaround.

I'd like to check for the existence of a repository variable prior to creating/updating that variable. I would expect a 404 response. However, when running the examples below I get a `Error: Unhandled error: HttpError: Not Found` message and the step dies - even though I want the 404 result value. Is there a way to get that value without the step erroring out? I tried `continue-on-error` but that doesn't work either since it's the `github-script` that's dying. Hopefully this is a simple documentation thing I missed. Thank you!

Step calling the script
```
- uses: actions/github-script@v6
        with:
          github-token: ${{ secrets.myPAT }}            
          script: |
            const script = require('./.github/scripts/createOrUpdateRepoVariable.js')
            await script({ github, context })
```

The script being called
```
module.exports = async ({ context, github, name, value }) => {
  const { status } = await github.request('GET /repos/{owner}/{repo}/actions/variables/{name}', {
    owner: context.repo.owner,
    repo: context.repo.repo,
    name: 'DOES_NOT_EXIST'
  })
  
  console.log(status)
}
```

---

The output logs when the step fails.
```
RequestError [HttpError]: Not Found
Error: Unhandled error: HttpError: Not Found
    at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:6842:21
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async module.exports (my stuff)
    at async eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15143:16), <anonymous>:4:1)
    at async main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15236:20) {
  status: 404,
  response: {
    url: 'my-repo-url',
    status: 404,
    headers: {
      'a-bunch-of-headers'
    },
    data: {
      message: 'Not Found',
      documentation_url: 'https://docs.github.com/rest/actions/variables#get-a-repository-variable'
    }
  },
  request: {
    method: 'GET',
    url: 'my-url',
    headers: {
      accept: 'application/vnd.github.v3+json',
      'user-agent': 'actions/github-script octokit-core.js/3.6.0 Node.js/16.16.0 (linux; x64)',
      authorization: 'token [REDACTED]'
    },
    request: { agent: [Agent], hook: [Function: bound bound register] }
  }
}
```

**Comment by joshft91** [comment, marked as answer]
I was messing around with this more and found this as a potential workaround/solution by catching  thrown error and returning the response object. Is there a better/cleaner way?

```
const response = await github.request('GET /repos/{owner}/{repo}/actions/variables/{name}', {
    owner: context.repo.owner,
    repo: context.repo.repo,
    name
  }).catch(response => response)
```

**Comment by joshmgross** [comment]
👋 Hey @joshft91,

I think your solution is on the right track, you want to catch any errors being thrown by Octokit for errors.

Here's another example in case you wanted to add some logic in the case of errors:
```ts
module.exports = async ({ context, github, name, value }) => {
  let status;
  try {
    const { respStatus } = await github.request('GET /repos/{owner}/{repo}/actions/variables/{name}', {
      owner: context.repo.owner,
      repo: context.repo.repo,
      name: 'DOES_NOT_EXIST'
    })
    status = respStatus
  } catch (e) {
    status = e.status
  }

  console.log(status)
}
```

**Reply by joshft91** [reply]
Thanks for the response!  Glad to know I'm on the right track. Looking at your example, could `.then(...)` be used to get the same status as yours? I think it would, thoughI haven't tested it. Either way, thanks for the examples. I think I am good to go.

```
module.exports = async ({ context, github, name, value }) => {
    const status = await github.request('GET /repos/{owner}/{repo}/actions/variables/{name}', {
      owner: context.repo.owner,
      repo: context.repo.repo,
      name: 'DOES_NOT_EXIST'
    }).then(response => response.status)
      .catch(response => response.status)

  console.log(status)
}
```
